web/src/client/mixins.js

```diff
@@ -222,13 +226,16 @@ const WithProgress = (superclass, object_path) =>
      * @return {import ("./dbus").RemoveFn} function to disable the callback
      */
     onProgressChange(handler) {
-      return this.client.onObjectChanged(object_path, PROGRESS_IFACE, (changes) => {
-        const { TotalSteps, CurrentStep, Finished } = changes;
-        if (TotalSteps === undefined && CurrentStep === undefined && Finished === undefined) {
-          return;
+      return this.client.onEvent("Progress", (progress) => {
+        if (progress?.service === service_name) {
```

Comment:
is it possible to get undefined service? Is it worth logging it?

Reply:
It should not happen, but I wanted to avoid a crash as we cannot control what comes through the wire.